### PR TITLE
Update developer package specs for workbench requirements

### PR DESCRIPTION
**Pull request by martyngigg**
Description of work.

Adds Qt5 plus other workbench requirements to the developer spec file and debian control files.

**To test:**

* Code review

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec [hunk 1]
```diff
@@ -47,7 +47,10 @@ Requires: numpy
 Requires: OCE-devel
 Requires: poco-devel >= 1.4.6
 Requires: PyQt4-devel
+Requires: python-qt5-devel
+Requires: python-QtPy
```
Comment (peterfpeterson) [inline buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec]: Missing the python3 version below

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

**Review by peterfpeterson — APPROVED** [review]

**Comment by rosswhitfield** [comment]
`qscintilla-qt5-devel`

**Comment by peterfpeterson** [comment]
I must have had that installed already. This is approved to merge.

**Comment by rosswhitfield** [comment]
Do we also need `libqt5scintilla2-dev` for Ubuntu?

**Comment by martyngigg** [comment]
Build failures are unrelated to this changes.

**Pull request merged**